### PR TITLE
Autoformatting

### PR DESCRIPTION
**Pull request by tsa96**
Did this a few weeks ago and forgot to PR. Adds Prettier autoformatting, as an npm script and a CI job.

Don't love making this an npm package, but devs won't necessarily need to run `npm install` for any of this if they have a Prettier extension for their editor and use it. But it's there if desired, and is needed for CI job anyway.

### CODE REVIEW

File: shared/events.d.ts [hunk 1]
```diff
@@ -4,88 +4,88 @@
...
+	ScrollRight: () => void;
+	ScrollUp: () => void;
+	SettingsNavigateToPanel: (category: string, panel: Panel) => void;
+	ShowCenterPrintText: (message: string, priority: unknown) => void;
+	ShowContentPanel: () => void;
+	ShowVoteContextMenu: () => void;
+	StaticHudMenu_EntrySelected: (panel: Panel) => void;
+	UnloadLoadingScreenAndReinit: () => void;
+	AchievementInfoLoaded: () => void;
+	AchievementEarned: (player_index: number, achievement_index: number) => void;
```
Comment (koerismo) [inline shared/events.d.ts:90]: is there anything we can do so it doesn't totally fuck up the formatting of this file?

File: p2ce/achievements.d.ts [hunk 2]
```diff
@@ -1,17 +1,15 @@
...
+	index: number; // Index of the achievement (for use with the API)
+	achieved: boolean; // True if this has been achieved
+	available: boolean; // True if this is available
+	hide_until_achieved: boolean; // True if this is a "hidden" achievement. It should not be displayed in its full glory until achieved
 	flags: number;
-	count: number;						// Current count, if a stat is involved with this achievement. The achievement is granted when count >= goal
-	goal: number;						// Stat goal for this achievement
+	count: number; // Current count, if a stat is involved with this achievement. The achievement is granted when count >= goal
+	goal: number; // Stat goal for this achievement
 }
```
Comment (koerismo) [inline p2ce/achievements.d.ts:10]: not a huge fan of what it's doing with the comments here (and elsewhere), they were somewhat easier to read as they were.

**Review by koerismo — CHANGES_REQUESTED** [review]

File: shared/events.d.ts (hunk 1 above)
Comment (tsa96) [reply shared/events.d.ts:90]: No, prettier will always get rid of indentation like this, personally I prefer it prettier's way, find it easier to read and edit, if you really can't stand it we wouldn't be able to use prettier (happy to discuss that).

File: p2ce/achievements.d.ts (hunk 2 above)
Comment (tsa96) [reply p2ce/achievements.d.ts:10]: Oh yeah, I can convert to single line `/** */` comments.

**Review by koerismo — APPROVED** [review]

**Comment by tsa96** [comment]
Closing as per internal discussion

**Comment by tsa96** [comment]
Will leave branch in case we change our mind in the future

**Pull request closed without merging**